src/Wordpress/CustomPostType/Timeframe.php

```diff
@@ -966,6 +966,71 @@ public function setCustomColumnsData( $column, $post_id ) {
 		}
 	}
 
+	public function setCustomColumnSortOrder( \WP_Query $query ) {
+		parent::setCustomColumnSortOrder( $query );
+		//TODO: Refactor this so we don't have to repeat ourselves in every inheritance
+		if ( ! is_admin() || ! $query->is_main_query() || $query->get( 'post_type' ) !== static::$postType ) {
+			return;
+		}
+		switch ($query->get( 'orderby' )) {
+			case 'item-id':
+				add_filter('posts_join', function ($join) {
+					global $wp_query, $wpdb;
+
+					if ( ! empty( $wp_query->query_vars['orderby'] ) && $wp_query->query_vars['orderby'] === \CommonsBooking\Model\Timeframe::META_ITEM_ID ) {
+						$join .= "LEFT JOIN $wpdb->postmeta joined_meta_items "
+						         . "ON $wpdb->posts.ID = joined_meta_items.post_id AND joined_meta_items.meta_key = '" . \CommonsBooking\Model\Timeframe::META_ITEM_ID . "' ";
+						$join .= "JOIN $wpdb->posts joined_items ON joined_meta_items.meta_value = joined_items.ID ";
+					}
+
+					return $join;
+				});
+				add_filter( 'posts_orderby', function ( $orderby ) {
+					global $wp_query;
+
+					if ( ! empty( $wp_query->query_vars['orderby'] ) && $wp_query->query_vars['orderby'] === \CommonsBooking\Model\Timeframe::META_ITEM_ID ) {
+						$orderby = 'joined_items.post_title ' . $wp_query->query_vars['order'];
+					}
+
+					return $orderby;
+				});
+				break;
+			case 'location-id':
+				add_filter('posts_join', function ($join) {
+					global $wp_query, $wpdb;
+
+					if ( ! empty( $wp_query->query_vars['orderby'] ) && $wp_query->query_vars['orderby'] === \CommonsBooking\Model\Timeframe::META_LOCATION_ID ) {
+						$join .= "LEFT JOIN $wpdb->postmeta joined_meta_locations "
+						         . "ON $wpdb->posts.ID = joined_meta_locations.post_id AND joined_meta_locations.meta_key = '" . \CommonsBooking\Model\Timeframe::META_LOCATION_ID . "' ";
+						$join .= "JOIN $wpdb->posts joined_locations ON joined_meta_locations.meta_value = joined_locations.ID ";
+					}
+
+					return $join;
+				});
+				add_filter( 'posts_orderby', function ( $orderby ) {
+					global $wp_query;
+
+					if ( ! empty( $wp_query->query_vars['orderby'] ) && $wp_query->query_vars['orderby'] === \CommonsBooking\Model\Timeframe::META_LOCATION_ID ) {
+						$orderby = 'joined_locations.post_title ' . $wp_query->query_vars['order'];
+					}
+
+					return $orderby;
+				});
+				break;
+			case 'type':
+				$value = $query->get( 'orderby' );
+				$query->set( 'meta_key', $value );
```

Comment:
$value ist hier konstant gleich 'type'. Zeile 1021 sollte daher weg und in 1022 statt $value einfach 'type' verwendet werden.